diffrax/_integrate.py

```diff
@@ -803,6 +804,9 @@ def _promote(yi):
                 "`UnsafeBrownianPath` cannot be used with adaptive step sizes."
             )
 
+    if isinstance(solver, KLSolver):
+        y0 = (y0, 0.0)
+        y0 = jtu.tree_map(_promote, y0)
     # Normalises time: if t0 > t1 then flip things around.
```

Comment:
I don't think this is a thing we should put in `_integrate.py`. Rather, we should have some kind of
```python
def kl_foo_bar(...):
    ...
```
function, which can be used as
```python
solver, y0 = kl_foo_bar(...)
diffeqsolve(solver=solver, y0,=y0 ...)
```

Reply:
Would you want y0 out of `kl_foo_bar` to be specific `KL y0` class or would a tuple be fine? Also, would we want to modify the y0 and the solver at the same time or have a KL solver and a KL y0 transformer?